mlir/lib/IR/MLIRContext.cpp

```diff
@@ -188,7 +188,15 @@ class MLIRContextImpl {
 
   /// This is a sorted container of registered operations for a deterministic
   /// and efficient `getRegisteredOperations` implementation.
-  SmallVector<RegisteredOperationName, 0> sortedRegisteredOperations;
+  SmallVector<std::pair<StringRef, RegisteredOperationName>, 0>
+      sortedRegisteredOperations;
+
+  /// This stores the transformed operations when calling
+  /// `getRegisteredOperations`.
+  SmallVector<RegisteredOperationName, 0> transformedOperations;
+
+  /// This returns the number of registered operations for a given dialect.
+  llvm::DenseMap<StringRef, size_t> getCountByDialectName;
```

Comment:
Why not have this stored in the dialect itself? That'd remove an extra map.

Also, nit: please drop the llvm:: (DenseMap is already exported in the mlir namespace).

Reply:
This isn't done right now?
The suggestion I believe is to add a "int operationCount = 0;` in the `class Dialect` where the counter would live and remove this map entirely from the Context.

Reply:
Fixed.